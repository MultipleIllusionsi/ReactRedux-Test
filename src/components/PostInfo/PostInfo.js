import React from "react";
import PropTypes from "prop-types";

import "./PostInfo.scss";

const PostInfo = ({ data }) => (
  <div className="info">
    <h3>{data.title}</h3>
    <p>{data.body}</p>
    <p>#{data.id}</p>
  </div>
);

PostInfo.propTypes = {
  data: PropTypes.object.isRequired
};

export default PostInfo;
