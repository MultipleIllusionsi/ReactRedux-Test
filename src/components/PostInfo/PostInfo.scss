.info {
    width: 75%;

    & h3 {
        padding-bottom: 1rem;
        font-size: 3rem;
        border-bottom: 1px solid #a4b0be;
    }

    & p {
        font-size: 1.7rem;
        width: 70%;
        margin: 2rem 0 0;
    }
}

.post-info {
    padding: 3rem;
    border: 1px solid rgb(110, 106, 106);
    display: flex;
    justify-content: space-between;
    box-shadow: 0 0.2rem 0.3rem 0 black;
    transition: 0.2s all ease-out;

    &:hover {
        box-shadow: 0 0.5rem 0.3rem 0.2rem black;
        transform: translateY(-0.3rem);
    }
}