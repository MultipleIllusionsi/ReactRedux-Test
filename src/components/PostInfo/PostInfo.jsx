import React from "react";

import "./PostInfo.scss";

const PostInfo = ({ data }) => (
  <div className="info">
    <h3>{data.title}</h3>
    <p>{data.body}</p>
    <p>#{data.id}</p>
  </div>
);
export default PostInfo;
