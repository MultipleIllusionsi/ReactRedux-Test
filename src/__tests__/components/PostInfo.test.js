import { shallow } from "enzyme";
import React from "react";
import PostInfo from "../../components/PostInfo/PostInfo";

it("expect to render PostInfo component", () => {
  const post = {
    userId: 2,
    id: 10,
    body: "some text",
    title: "some title"
  };
  const component = shallow(<PostInfo data={post} />);
  expect(component.debug()).toMatchSnapshot();
});
