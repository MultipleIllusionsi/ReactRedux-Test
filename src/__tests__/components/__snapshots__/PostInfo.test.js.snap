// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render PostInfo component 1`] = `
"<div className=\\"info\\">
  <h3>
    some title
  </h3>
  <p>
    some text
  </p>
  <p>
    #
    10
  </p>
</div>"
`;
